osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -285,12 +285,24 @@ private void criteriaChanged(FilterCriteria criteria)
             filterDebounce = Scheduler.AddDelayed(() => carousel.Filter(criteria), filter_delay);
         }
 
+        private void updateMatchesNote()
+        {
+            int count = carousel.MatchedBeatmapsCount;
+
+            if (count == 1)
+                filterControl.InformationalNote = @"1 match";
+            else
+                filterControl.InformationalNote = $@"{count:#,0} matches";
+        }
+
         #endregion
 
         protected override void Update()
         {
             base.Update();
+
             detailsArea.Height = wedgesContainer.DrawHeight - titleWedge.LayoutSize.Y - 4;
+            updateMatchesNote();
```

Comment:
wait hold up, why is this in `Update` with no debouncing at all? it's going to cause string interpolation allocation overheads *every frame*??